src/parsing/staticallyCompileXPath.ts

```diff
@@ -44,69 +52,153 @@ export default function staticallyCompileXPath(
 				functionNameResolver
 		  );
 
-	const executionSpecificStaticContext = new ExecutionSpecificStaticContext(
-		namespaceResolver,
-		variables,
-		defaultFunctionNamespaceURI,
-		functionNameResolver
-	);
-	const rootStaticContext = new StaticContext(executionSpecificStaticContext);
-
-	let expression: Expression;
-
 	if (fromCache !== null) {
-		expression = fromCache.expression;
+		return {
+			state: CACHE_STATE.COMPILED,
+			expression: fromCache.expression,
+		};
 	} else {
 		// We can not use anything from the cache, parse + compile
 		const ast = parseExpression(xpathString, compilationOptions);
-		const context = new AnnotationContext(rootStaticContext);
-		if (compilationOptions.annotateAst) {
-			annotateAst(ast, context);
-		}
 
-		const mainModule = astHelper.getFirstChild(ast, 'mainModule');
-		if (!mainModule) {
-			// This must be a library module
-			throw new Error('Can not execute a library module.');
-		}
+		return {
+			state: CACHE_STATE.PARSED,
```

Comment:
This is a "cache state", but the value is not actually cached?

Reply:
The value is not cached directly, but it will be afterwards I think